src/tsf/TfCtxtComp.h

```diff
@@ -21,32 +21,15 @@ Revision History:
 
 #pragma once
 
-#include "StructureArray.h"
-
-/////////////////////////////////////////////////////////////////////////////
-// CCompString
-
-class CCompString : public CComBSTR
-{
-public:
-    CCompString() : CComBSTR() { };
-    CCompString(__in_ecount(dwLen) LPWSTR lpsz, DWORD dwLen) : CComBSTR(dwLen, lpsz) { };
-    virtual ~CCompString() { };
-};
-
 /////////////////////////////////////////////////////////////////////////////
 // CCompAttribute
 
-class CCompAttribute : public CStructureArray<BYTE>
+class CCompAttribute : public std::vector<BYTE>

```

Comment:
Good call on dropping the custom array for a vector.

Reply:
It seems that `CCompAttribute` is not used at all.

Reply:
Then drop it completely! Even better!